### PR TITLE
feat: support sprig library for trigger templates

### PR DESCRIPTION
**Pull request by gODeaLoAple**
# Add sprig library templates

https://pkg.go.dev/github.com/masterminds/sprig#section-readme

Also filter some functions in sprig to prevent possible unsafety/insecure usages

### CODE REVIEW

File: templating/templating.go [hunk 1]
```diff
@@ -43,6 +44,19 @@ type trigger struct {
 	Name string `json:"name"`
 }
 
+type FuncMap = template.FuncMap
```
Comment (kissken) [inline templating/templating.go:47]: new type is extra, i think

File: templating/templating.go [hunk 2]
```diff
@@ -64,12 +78,198 @@ func Populate(name, description string, events []Event) (desc string, err error)
 		"stringsSplit":      strings.Split,
 	}
 
+	sprigFuncMap := FilterKeys(sprig.FuncMap(), []string {
```
Comment (kissken) [inline templating/templating.go:81]: better call FilterKeys only once for start, not with call "Populate"

Comment (gODeaLoAple) [reply templating/templating.go]: I moved out this and funcsMap out of "Populate" scope

File: templating/templating.go [hunk 3]
```diff
@@ -43,6 +45,214 @@ type trigger struct {
 	Name string `json:"name"`
 }
 
+func FilterKeys(source template.FuncMap, keys []string) template.FuncMap {
+	result := template.FuncMap{}
+	for _, key := range keys {
+		value, found := source[key]
+		if found {
```
Comment (Tetrergeru) [inline templating/templating.go:52]:
```suggestion
		if value, ok := source[key], ok {
```

**Review by Tetrergeru — APPROVED** [review]

File: templating/templating.go [hunk 4]
```diff
@@ -43,6 +45,214 @@ type trigger struct {
 	Name string `json:"name"`
 }
 
+func FilterKeys(source template.FuncMap, keys []string) template.FuncMap {
```
Comment (kissken) [inline templating/templating.go:48]: and let's fix linter and add godoc for this func or make func private better FilterKeys=> filterKeys :)

File: templating/templating.go (hunk 3 above)
Comment (gODeaLoAple) [reply templating/templating.go:52]: Does not work for me
image

File: templating/templating.go (hunk 4 above)
Comment (gODeaLoAple) [reply templating/templating.go]: done

File: templating/templating.go (hunk 3 above)
Comment (kissken) [reply templating/templating.go:52]: if value, ok := source[key], ok { => if value, ok := source[key]**;** ok {

Comment (kissken) [reply templating/templating.go:52]: not ,
use ; in this :)

Comment (Tetrergeru) [reply templating/templating.go:52]: Sorry, it was a typo

**Review by Tetrergeru — APPROVED** [review]

File: templating/templating.go (hunk 3 above)
Comment (gODeaLoAple) [reply templating/templating.go]: fixed

**Review by kissken — APPROVED** [review]

**Pull request merged**